editor/meta-boxes/meta-boxes-area/index.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import { isEqual } from 'lodash';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import { Component } from '@wordpress/element';
+import { Spinner } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import './meta-box-area.scss';
+import { handleMetaBoxReload, metaBoxStateChanged, metaBoxLoaded } from '../../actions';
+import { getMetaBox, isSavingPost } from '../../selectors';
+
+class MetaBoxesArea extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			loading: true,
+		};
+		this.originalFormData = [];
+		this.bindNode = this.bindNode.bind( this );
+		this.checkState = this.checkState.bind( this );
+	}
+
+	bindNode( node ) {
+		this.node = node;
+	}
+
+	componentDidMount() {
+		this.mounted = true;
+		this.fetchMetaboxes();
+	}
+
+	componentWillUnmout() {
+		this.mounted = false;
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.isUpdating && ! this.props.isUpdating ) {
+			this.setState( { loading: true } );
+			const { location } = nextProps;
+			const headers = new window.Headers();
+			const fetchOptions = {
+				method: 'POST',
+				headers,
+				body: new window.FormData( this.form ),
+				credentials: 'include',
+			};
+			const request = window.fetch( addQueryArgs( window._wpMetaBoxUrl, { meta_box: location } ), fetchOptions );
+			this.onMetaboxResponse( request, false );
+		}
+	}
+
+	fetchMetaboxes() {
+		const { location } = this.props;
+		const request = window.fetch( addQueryArgs( window._wpMetaBoxUrl, { meta_box: location } ), { credentials: 'include' } );
+		this.onMetaboxResponse( request );
+	}
+
+	onMetaboxResponse( request, initial = true ) {
+		request.then( ( response ) => response.text() )
+			.then( ( body ) => {
+				if ( ! this.mounted ) {
+					return;
+				}
+				jQuery( this.node ).html( body );
```

Comment:
Why do we use jQuery for this? `this.node.innerHTML = body` ?

(To earlier point that we don't have an explicit dependency on jQuery for the editor)

Reply:
We need to run the scripts contained in `body`. jQuery does this pretty well. I tried recreating the behavior myself, it "mostly" worked but not 100%. Is there a specific lib to do so?

Reply:
I would think `innerHTML` should be running any newly injected scripts.

Reply:
Can't find official documentation about this, but it's not running the injected scripts. See https://stackoverflow.com/questions/2592092/executing-script-elements-inserted-with-innerhtml